src/Aspire.Hosting/Dcp/DcpHostService.cs

```diff
@@ -257,6 +269,108 @@ private void EnsureDockerIfNecessary()
+        IAsyncDisposable? processDisposable = null;
+
+        try
+        {
+            // Init
+            Task<ProcessResult> task;
+            var outputStringBuilder = new StringBuilder();
+
+            string? dcpExePath = _dcpOptions.CliPath;
+            if (!File.Exists(dcpExePath))
+            {
+                throw new FileNotFoundException($"The Aspire application host is not installed at \"{dcpExePath}\". The application cannot be run without it.", dcpExePath);
+            }
+
+            // Run `dcp version`
+            var processSpec = new ProcessSpec(dcpExePath)
+            {
+                Arguments = $"version",
+                OnOutputData = s => outputStringBuilder.Append(s),
+            };
+
+            (task, processDisposable) = ProcessUtil.Run(processSpec);
+
+            await task.WaitAsync(TimeSpan.FromSeconds(WaitTimeForDcpVersionCommandInSeconds), cancellationToken).ConfigureAwait(false);
+
+            // Parse the output as JSON
+            var output = outputStringBuilder.ToString();
+            if (output == string.Empty)
+            {
+                return; // Best effort
+            }
+
+            var dcpVersionString = JsonNode.Parse(outputStringBuilder.ToString())?.AsObject()["version"]?.GetValue<string>();
+
+            if (dcpVersionString == null
+                || dcpVersionString == string.Empty
+                || dcpVersionString == "dev")
+            {
+                // If empty, null, or a dev version, pass
+                return;
+            }
+
+            // Early DCP versions (e.g. preview 1) have a +x at the end of their version string, e.g. 0.1.42+5,
+            // which does not parse. Strip off anything like that.
+            dcpVersionString = VersionRegex().Replace(dcpVersionString, string.Empty);
+
+            Version? dcpVersion;
+            if (Version.TryParse(dcpVersionString, out dcpVersion))
+            {
+                if (dcpVersion < DcpVersion.MinimumVersionInclusive)
+                {
+                    Console.Error.WriteLine(string.Format(
+                        CultureInfo.InvariantCulture,
+                        Resources.DcpVersionCheckTooLowMessage
+                    ));
+                    Environment.Exit((int)DcpVersionCheckFailures.DcpVersionIncompatible);
+                }
+                else if (dcpVersion >= DcpVersion.MaximumVersionExclusive)
+                {
+                    Console.Error.WriteLine(string.Format(
+                        CultureInfo.InvariantCulture,
+                        Resources.DcpVersionCheckTooHighMessage,
+                        DcpVersion.MinimumVersionInclusive.ToString()
+                    ));
+                    Environment.Exit((int)DcpVersionCheckFailures.DcpVersionIncompatible);
+                }
+            }
+        }
+        catch (Exception ex) when (ex is not DistributedApplicationException)
+        {
+            Console.Error.WriteLine(string.Format(
```

Comment:
Why is this writing to the Console directly, and not logging an error using `ILogger`?

Reply:
Ahh, we keep adding these 😢